testutil/validatormock/synccomm.go

```diff
@@ -144,19 +146,31 @@ func (s *SyncCommMember) PrepareSlot(ctx context.Context, slot eth2p0.Slot) erro
 	return nil
 }
 
-// Message submits sync committee messages at 1/3rd into the slot.
+// Message submits sync committee messages at 1/3rd into the slot. It also sets the beacon block root for the slot.
 func (s *SyncCommMember) Message(ctx context.Context, slot eth2p0.Slot) error {
 	wait(ctx, s.dutiesOK)
-	return submitSyncMessage(ctx, s.eth2Cl, slot, s.signFunc, s.duties)
+
+	blockRoot, err := s.eth2Cl.BeaconBlockRoot(ctx, "head")
+	if err != nil {
+		return err
+	}
+
+	err = submitSyncMessages(ctx, s.eth2Cl, slot, blockRoot, s.signFunc, s.duties)
+	if err != nil {
+		return err
+	}
+
+	s.blockRoots[slot] = blockRoot
```

Comment:
you need to add a synchronisation channel so Aggregate waits for this to be populated

Reply:
yeah, missed that